src/sema/ast.rs

```diff
@@ -649,7 +650,8 @@ pub struct Contract {
     pub initializer: Option<usize>,
     pub default_constructor: Option<(Function, usize)>,
     pub cfg: Vec<ControlFlowGraph>,
-    pub code: Vec<u8>,
+    /// Compiled program. Only available after emit.
+    pub code: OnceCell<Vec<u8>>,
```

Comment:
Why do we need an `OnceCell` here?

Reply:
In codegen, we don't have `mut` access to the Namespace. We generate the cfg and assume the namespace is read-only. However, we are storing the compiled code there to avoid double compilation.